Add connector for Rokari Comics (#8071)

2016mehrab opened the pull request:
Closes #8071

This Pull Request introduces a new connector to add support for the Rokari Comics website.

**What this PR does:**
* Adds `RokariComics.mjs` as a new source connector.
* Enables Browse and downloading manga/webtoons from `https://rokaricomics.com`.

**How it works:**
This connector implements the following core HakuNeko methods:
* `_getMangas()`
* `_getChapters(manga)`
* `_getPages(chapter)`
* `_getMangaFromURI(uri)`

2016mehrab committed:
Add connector for Rokari Comics (#8071)

MikeZeDev commented on src/web/mjs/connectors/RokariComics.mjs:
```diff
@@ -0,0 +1,68 @@
+        super.id = "rokaricomics";
+        super.label = "Rokari Comics";
+        this.tags = ["webtoon", "english"];
+        this.url = "https://rokaricomics.com";
```
* All of this is way too complicated. This website is using what we call "Mangastream" template. You can just extend it and its settled. There are any samples if you want.

* You are lucky that its an easy website because we'd like people to add website to Haruneko instead of Hakuneko.
Hakuneko PR needs to be reviewed & then website added to Haruneko too meaning its double work. I'll let it pass dont worry.

2016mehrab replied:
hello @MikeZeDev ,thanks i'll do that. I tried running haruneko using lutris but that didn't work (If you know a way to run it in linux, please let me know). Anyway, I am a bit confused as the RokariComics source shows 'wp-content/themes/mangareader' and there's also a 'MangaReaderCMS' template in hakuneko. is 'MangaReaderCMS' a different thing?

MikeZeDev replied:
_Idk_ about linux tbh. We still dont have a build but perhaps you could run it from source.

Mangareader is too generic name. In Hakuneko, websites that looks like Rokari use "MangaStream" template and has nothing to do with MangareaderCMS (sample website : h--ps://www.mangadenizi.net/ )

I cant tell the reason for that name, it was chosen way before i joined.

2016mehrab replied:
Hi @MikeZeDev , I've rewritten the RokariComics connector using the template and have pushed the changes.

MikeZeDev reviewed (changes requested):
Suggestion : rewrite it using Mangastream template

2016mehrab committed:
refactor(rokaricomics): Rewrite RokariComics to extend WordPressMangastream

Rewrote the Rokari Comics connector to extend the `WordPressMangastream` template, following review feedback.
The `_getChapters` method has been overridden to filter out premium chapters by checking for valid `href` attributes.
Also includes the `links.login` property.

MikeZeDev commented on src/web/mjs/connectors/RokariComics.mjs:
```diff
@@ -1,68 +1,32 @@
+        let request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryChapters);
+        return data
+        // filtering out premium chapters
```
dont filter premium chapters.

2016mehrab replied:
Premium chapters don't include any links, can anything be done about that?

MikeZeDev replied:
Ah, mb. Override this.queryChapter in constructor with a CSS selector that omit premium chapters (without href)

```js
        this.queryChapters = 'div#chapterlist ul li div.eph-num a[href]';
```
<img width="657" height="325" alt="image" src="https://github.com/user-attachments/assets/be09f645-f97d-45f5-816a-b3fc5f534516" />

2016mehrab committed:
refactor(RokariComics): Filter out premium chapters by checking for href attribute

- Added direct chapter query selector
- Removed unnecessary async function
